src/OpenTelemetry.Api/Trace/SpanAttributes.cs

```diff
@@ -0,0 +1,140 @@
+﻿// <copyright file="SpanAttributes.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A class that represents the span attributes. Read more here https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md#attributes.
+    /// </summary>
+    public class SpanAttributes
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanAttributes"/> class.
+        /// </summary>
+        public SpanAttributes()
+        {
+            this.Attributes = new ActivityTagsCollection();
+        }
+
+        internal ActivityTagsCollection Attributes { get; }
+
+        /// <summary>
+        /// Adds the elements of the specified collection to the <see cref="SpanAttributes"/>.
+        /// </summary>
+        /// <param name="initialAttributes">The collection whose elements should be added.</param>
+        public void AddRange(IEnumerable<KeyValuePair<string, object>> initialAttributes)
```

Comment:
We shouldn't accept string, object. Spec don't allow it.

Reply:
so, maybe we should remove addrange and maintain only the constructor and add methods, right?

Reply:
just removed.